hack/demo/generate-asciidemo.sh

```diff
@@ -13,45 +13,79 @@ function check_prereq() {
     fi
 }
 
-function cleanup() {
-    if [ -d $WKDIR ]
-    then
+cleanup() {
+    if [[ -n "${WKDIR-}" && -d $WKDIR ]]; then
         rm -rf $WKDIR
     fi
 }
 
-function usage() {
-    echo "$0 [options]"
-    echo "where options is"
-    echo " h  help (this message)"
+usage() {
+    echo "$0 [options] <demo_script>"
+    echo ""
+    echo "example:"
+    echo "  # Generate asciinema demo described by gzip-demo-script.sh to gzip-demo-script.cast"
+    echo "  $0 gzip-demo-script.sh"
+    echo ""
+    echo "  # Generate asciinema demo described by demo-script.sh to catalogd-demo.cast"
+    echo "  $0 -n catalogd-demo demo-script.sh"
+    echo ""
+    echo "  # Generate and upload catalogd-demo.cast"
+    echo "  $0 -u -n catalogd-demo demo-script.sh"
+    echo ""
+    echo "options:"
```

Comment:
nit: I'm not sure if there is a governing standard, but I generally see (and expect) options before examples.

Reply:
I don't have strong feelings here. I'll change it up (and found a couple of more things to fix)